src/Microsoft.Net.Http.Headers/EntityTagHeaderValue.cs

```diff
@@ -103,7 +103,7 @@ public override bool Equals(object obj)
             }
 
             // Since the tag is a quoted-string we treat it case-sensitive.
-            return ((_isWeak == other._isWeak) && (string.CompareOrdinal(_tag, other._tag) == 0));
```

Comment:
Fascinating, didn't know the defaults can vary in such ways. I guess it's always better to be on the safe side and just be explicit in what you want to do haha.